server/sonar-server/src/main/java/org/sonar/server/component/ws/ProvisionedProjectsAction.java

```diff
@@ -0,0 +1,129 @@
+
+  public ProvisionedProjectsAction(DbClient dbClient) {
+    this.dbClient = dbClient;
+  }
+
+  @Override
+  public void define(WebService.NewController controller) {
+    WebService.NewAction action = controller
+      .createAction("provisioned")
+      .setDescription(
+        "Get the list of provisioned projects.<br /> " +
+          "Require admin role.")
+      .setSince("5.2")
+      .setResponseExample(Resources.getResource(getClass(), "projects-example-provisioned.json"))
+      .setHandler(this)
+      .addPagingParams(100)
+      .addFieldsParam(Arrays.asList(POSSIBLE_FIELDS));
+
+    action
+      .createParam(Param.TEXT_QUERY)
+      .setDescription("UTF-8 search query")
+      .setExampleValue("sonar");
+  }
+
+  @Override
+  public void handle(Request request, Response response) throws Exception {
+    UserSession.get().checkGlobalPermission(UserRole.ADMIN, "You need admin rights.");
+    SearchOptions options = new SearchOptions().setPage(
+      request.mandatoryParamAsInt(Param.PAGE),
+      request.mandatoryParamAsInt(Param.PAGE_SIZE)
+      );
+    List<String> desiredFields = desiredFields(request);
+    String query = request.param(Param.TEXT_QUERY);
+
+    DbSession dbSession = dbClient.openSession(false);
+    List<ComponentDto> projects;
+    int nbOfProjects;
+    try {
+      projects = dbClient.componentDao().selectProvisionedProjects(dbSession, options, query);
+      nbOfProjects = dbClient.componentDao().countProvisionedProjects(dbSession, query);
+    } finally {
+      MyBatis.closeQuietly(dbSession);
+    }
+
+    JsonWriter json = response.newJsonWriter().beginObject();
+    writeProjects(projects, json, desiredFields);
+    options.writeJson(json, nbOfProjects);
+    json.endObject().close();
+  }
+
+  private void writeProjects(List<ComponentDto> projects, JsonWriter json, List<String> desiredFields) {
+    json.name("projects");
+    json.beginArray();
+    for (ComponentDto project : projects) {
+      json.beginObject();
+      json.prop("uuid", project.uuid());
+      writeIfNeeded(json, "key", project.key(), desiredFields);
+      writeIfNeeded(json, "name", project.name(), desiredFields);
+      writeIfNeeded(json, "creationDate", project.getCreatedAt(), desiredFields);
+      json.endObject();
+    }
+    json.endArray();
+  }
+
+  private void writeIfNeeded(JsonWriter json, String fieldName, String value, List<String> desiredFields) {
+    if (desiredFields.contains(fieldName)) {
+      json.prop(fieldName, value);
+    }
+  }
+
+  private void writeIfNeeded(JsonWriter json, String fieldName, Date date, List<String> desiredFields) {
+    if (desiredFields.contains(fieldName)) {
+      json.propDate(fieldName, date);
+    }
+  }
+
+  private List<String> desiredFields(Request request) {
+    List<String> desiredFields = request.paramAsStrings(Param.FIELDS);
+    if (desiredFields == null) {
+      desiredFields = Arrays.asList(POSSIBLE_FIELDS);
+    }
+
+    return desiredFields;
```

Comment:
ok, so If I understand correctly, when field parameter exists in the request, there is no validation nor indirection on the field names?

This is hardly desirable:
1. this WS being public, we are not free anymore to change column names as we like
2. client can specify any column name, even if its internal, apparently only the supported ones are written to the response but those unsupported will still be retrieved from DB